addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py

```diff
@@ -348,6 +349,16 @@ def _import_fill_invoice_form(self, journal, tree, invoice_form, qty_factor):
                     invl_logs = self._import_fill_invoice_line_form(journal, invl_el, invoice_form, invoice_line_form, qty_factor)
                     logs += invl_logs
 
+        # ==== total tax (to avoid rounding mismatch) ====
+
+        xml_tax_amount_per_rate = {}
+        for tax_node in tree.findall('.//{*}ApplicableHeaderTradeSettlement/{*}ApplicableTradeTax'):
+            rate = tax_node.find('.//{*}RateApplicablePercent').text
+            amount = tax_node.find('.//{*}CalculatedAmount').text
+            xml_tax_amount_per_rate[float(rate)] = float(amount)
+
+        self._force_invoice_tax_lines(invoice_form.line_ids._records, xml_tax_amount_per_rate)
```

Comment:
To be honest, I don't like this PR. At this point of the code, we already run some methods to retrieve the taxes from the value on the xml. Here, you are doing quite the same thing again but in a very different way. I think it would be better to track from the lines a mapping percentage/fixed amount -> account.tax and then, at the end, use this mapping to force the tax amounts. Also, I think the whole code of `_force_invoice_tax_lines` is broken in case of multiple repartition lines on taxes or when generating a credit note, when dealing with a foreign currency...